File: sample/src/main/java/com/your_company/ui_bindings_sample/CouponsActivity.java
package com.your_company.ui_bindings_sample;

import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;

import com.nearit.ui_bindings.NearITUIBindings;
import com.your_company.ui_bindings_sample.factories.CouponFactory;

import it.near.sdk.reactions.couponplugin.model.Coupon;

/**
 * @author Federico Boschini
 */

public class CouponsActivity extends AppCompatActivity {

    CouponFactory couponFactory;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_coupons);
        couponFactory = new CouponFactory();
    }

    /*  COUPON LIST EXAMPLES    */

    public void onDefaultClicked(View view) {
        //  start an activity that shows a list of REAL coupons, with jagged borders
        startActivity(NearITUIBindings.getInstance(CouponsActivity.this)
                .couponListIntentBuilder()
                .jaggedBorders()
                .enableNetErrorDialog()
                .build());
    }

    public void onValidClicked(View view) {
        startActivity(NearITUIBindings.getInstance(CouponsActivity.this)
                .couponListIntentBuilder()
                .validCoupons()
                .build());
    }

    public void onInactiveClicked(View view) {
        startActivity(NearITUIBindings.getInstance(CouponsActivity.this)
                .couponListIntentBuilder()
                .inactiveCoupons()
                .build());
    }

    public void onExpiredClicked(View view) {
        startActivity(NearITUIBindings.getInstance(CouponsActivity.this)
                .couponListIntentBuilder()
                .expiredCoupons()
                .build());
    }

    public void onRedeemedClicked(View view) {
        startActivity(NearITUIBindings.getInstance(CouponsActivity.this)
                .couponListIntentBuilder()
                .redeemedCoupons()
                .build());
    }

    public void onValidAndInactiveClicked(View view) {
        startActivity(NearITUIBindings.getInstance(CouponsActivity.this)
                .couponListIntentBuilder()
                .validCoupons()
                .inactiveCoupons()
                .build());
    }

    public void onValidAndExpiredClicked(View view) {
        startActivity(NearITUIBindings.getInstance(CouponsActivity.this)
                .couponListIntentBuilder()
                .validCoupons()
                .expiredCoupons()
                .build());
    }

    public void onValidAndRedeemedClicked(View view) {
        startActivity(NearITUIBindings.getInstance(CouponsActivity.this)
                .couponListIntentBuilder()
                .validCoupons()
                .redeemedCoupons()
                .build());
    }

    public void onInactiveAndExpiredClicked(View view) {
        startActivity(NearITUIBindings.getInstance(CouponsActivity.this)
                .couponListIntentBuilder()
                .inactiveCoupons()
                .expiredCoupons()
                .build());
    }

    public void onInactiveAndRedeemedClicked(View view) {
        startActivity(NearITUIBindings.getInstance(CouponsActivity.this)
                .couponListIntentBuilder()
                .inactiveCoupons()
                .redeemedCoupons()
                .build());
    }

    public void onExpiredAndRedeemedClicked(View view) {
        startActivity(NearITUIBindings.getInstance(CouponsActivity.this)
                .couponListIntentBuilder()
                .expiredCoupons()
                .redeemedCoupons()
                .build());
    }

    public void onPlainCouponListClicked(View view) {
        //  start a plain activity that shows the fragment fullscreen
        //  please check it
        startActivity(new Intent(CouponsActivity.this, CouponListPlainActivity.class));
    }


    /*  COUPON DETAIL EXAMPLES    */

    public void onValidDetailClicked(View view) {
        //  In a real scenario the coupon is provided by the NearIT SDK
        Coupon validCoupon = couponFactory.getValidCoupon();
        startActivity(
                //  Basic example with a valid coupon
                NearITUIBindings.getInstance(CouponsActivity.this)
                        .couponIntentBuilder(validCoupon)
                        .build());
    }

    public void onInactiveDetailClicked(View view) {
        //  In a real scenario the coupon is provided by the NearIT SDK
        Coupon validCoupon = couponFactory.getInactiveCoupon();
        startActivity(
                //  Basic example with a not yet valid coupon
                NearITUIBindings.getInstance(CouponsActivity.this)
                        .couponIntentBuilder(validCoupon)
                        .build());
    }

    public void onExpiredDetailClicked(View view) {
        Coupon expiredCoupon = couponFactory.getExpiredCoupon();
        startActivity(
                //  Basic example with an expired coupon
                //  + enable tap outside the dialog to close it
                NearITUIBindings.getInstance(CouponsActivity.this)
                        .couponIntentBuilder(expiredCoupon)
                        .enableTapOutsideToClose()
                        .build());
    }

    public void onCustomIconDetailClicked(View view) {
        //  In a real scenario the coupon is provided by the NearIT SDK
        Coupon validCoupon = couponFactory.getValidCoupon();
        startActivity(
                //  Change the default icon placeholder
                NearITUIBindings.getInstance(CouponsActivity.this)
                        .couponIntentBuilder(validCoupon)
                        .setIconPlaceholderResourceId(R.drawable.my_custom_placeholder)
                        .build());
    }

    public void onPlainCouponClicked(View view) {
        //  start a plain activity that shows the fragment fullscreen
        //  please check it
        startActivity(new Intent(CouponsActivity.this, CouponPlainActivity.class));
    }
}

File: sample-kotlin/src/main/java/com/your_company/ui_bindings_kotlin_sample/CouponsActivity.kt
package com.your_company.ui_bindings_kotlin_sample

import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.View
import com.nearit.ui_bindings.NearITUIBindings
import com.your_company.ui_bindings_kotlin_sample.factories.CouponFactory

/**
* @author Federico Boschini
*/

class CouponsActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_coupons)
    }

    /*  COUPON LIST EXAMPLES    */

    @Suppress("UNUSED_PARAMETER")
    fun onDefaultClicked(view: View) {
        //  start an activity that shows a list of REAL coupons, with jagged borders
        startActivity(NearITUIBindings.getInstance(this@CouponsActivity)
                .couponListIntentBuilder()
                .setTitle("My coupon list")
                .jaggedBorders()
                .build())
    }

    @Suppress("UNUSED_PARAMETER")
    fun onValidClicked(view: View) {
        startActivity(NearITUIBindings.getInstance(this@CouponsActivity)
                .couponListIntentBuilder()
                .validCoupons()
                .build())
    }

    @Suppress("UNUSED_PARAMETER")
    fun onInactiveClicked(view: View) {
        startActivity(NearITUIBindings.getInstance(this@CouponsActivity)
                .couponListIntentBuilder()
                .inactiveCoupons()
                .build())
    }

    @Suppress("UNUSED_PARAMETER")
    fun onExpiredClicked(view: View) {
        startActivity(NearITUIBindings.getInstance(this@CouponsActivity)
                .couponListIntentBuilder()
                .expiredCoupons()
                .build())
    }

    @Suppress("UNUSED_PARAMETER")
    fun onRedeemedClicked(view: View) {
        startActivity(NearITUIBindings.getInstance(this@CouponsActivity)
                .couponListIntentBuilder()
                .redeemedCoupons()
                .build())
    }

    @Suppress("UNUSED_PARAMETER")
    fun onValidAndInactiveClicked(view: View) {
        startActivity(NearITUIBindings.getInstance(this@CouponsActivity)
                .couponListIntentBuilder()
                .validCoupons()
                .inactiveCoupons()
                .build())
    }

    @Suppress("UNUSED_PARAMETER")
    fun onValidAndExpiredClicked(view: View) {
        startActivity(NearITUIBindings.getInstance(this@CouponsActivity)
                .couponListIntentBuilder()
                .validCoupons()
                .expiredCoupons()
                .build())
    }

    @Suppress("UNUSED_PARAMETER")
    fun onValidAndRedeemedClicked(view: View) {
        startActivity(NearITUIBindings.getInstance(this@CouponsActivity)
                .couponListIntentBuilder()
                .validCoupons()
                .redeemedCoupons()
                .build())
    }

    @Suppress("UNUSED_PARAMETER")
    fun onInactiveAndExpiredClicked(view: View) {
        startActivity(NearITUIBindings.getInstance(this@CouponsActivity)
                .couponListIntentBuilder()
                .inactiveCoupons()
                .expiredCoupons()
                .build())
    }

    @Suppress("UNUSED_PARAMETER")
    fun onInactiveAndRedeemedClicked(view: View) {
        startActivity(NearITUIBindings.getInstance(this@CouponsActivity)
                .couponListIntentBuilder()
                .inactiveCoupons()
                .redeemedCoupons()
                .build())
    }

    @Suppress("UNUSED_PARAMETER")
    fun onExpiredAndRedeemedClicked(view: View) {
        startActivity(NearITUIBindings.getInstance(this@CouponsActivity)
                .couponListIntentBuilder()
                .expiredCoupons()
                .redeemedCoupons()
                .build())
    }

    @Suppress("UNUSED_PARAMETER")
    fun onPlainCouponListClicked(view: View) {
        //  start a plain activity that shows the fragment fullscreen
        //  please check it
        startActivity(Intent(this@CouponsActivity, CouponListPlainActivity::class.java))
    }

    /*  COUPON DETAIL EXAMPLES    */
    @Suppress("UNUSED_PARAMETER")
    fun onValidDetailClicked(view: View) {
        //  In a real scenario the coupon is provided by the NearIT SDK
        val validCoupon = CouponFactory.getValidCoupon()
        startActivity(
                //  Basic example with a valid coupon
                NearITUIBindings.getInstance(this@CouponsActivity)
                        .couponIntentBuilder(validCoupon)
                        .build())
    }

    @Suppress("UNUSED_PARAMETER")
    fun onInactiveDetailClicked(view: View) {
        //  In a real scenario the coupon is provided by the NearIT SDK
        val validCoupon = CouponFactory.getInactiveCoupon()
        startActivity(
                //  Basic example with a not yet valid coupon
                NearITUIBindings.getInstance(this@CouponsActivity)
                        .couponIntentBuilder(validCoupon)
                        .build())
    }

    @Suppress("UNUSED_PARAMETER")
    fun onExpiredDetailClicked(view: View) {
        val expiredCoupon = CouponFactory.getExpiredCoupon()
        startActivity(
                //  Basic example with an expired coupon
                //  + enable tap outside the dialog to close it
                NearITUIBindings.getInstance(this@CouponsActivity)
                        .couponIntentBuilder(expiredCoupon)
                        .enableTapOutsideToClose()
                        .build())
    }

    @Suppress("UNUSED_PARAMETER")
    fun onCustomIconDetailClicked(view: View) {
        //  In a real scenario the coupon is provided by the NearIT SDK
        val validCoupon = CouponFactory.getValidCoupon()
        startActivity(
                //  Change the default icon placeholder
                NearITUIBindings.getInstance(this@CouponsActivity)
                        .couponIntentBuilder(validCoupon)
                        .setIconPlaceholderResourceId(R.drawable.my_custom_placeholder)
                        .build())
    }

    @Suppress("UNUSED_PARAMETER")
    fun onPlainCouponClicked(view: View) {
        //  start a plain activity that shows the fragment fullscreen
        //  please check it
        startActivity(Intent(this@CouponsActivity, CouponPlainActivity::class.java))
    }
}